src/components/structures/SpaceHierarchy.tsx

```diff
@@ -362,7 +362,7 @@ export const showRoom = (cli: MatrixClient, hierarchy: RoomHierarchy, roomId: st
     // Don't let the user view a room they won't be able to either peek or join:
     // fail earlier so they don't have to click back to the directory.
     if (cli.isGuest()) {
-        if (!room.world_readable && !room.guest_can_join) {
+        if (!room?.world_readable && !room?.guest_can_join) {
             defaultDispatcher.dispatch({ action: "require_registration" });
             return;
         }
```

Comment:
I'm not familiar with this part of the code base, but it seems to me that before, if room was undefined, we'd throw an error on line 365 trying to access properties of the room (I'm presuming that they would be caught somewhere).

With this change, if room was undefined, wouldn't we now enter the `if` block on 365 because both sides of the && would evaluate to true? I think `?.something` evaluates to undefined if the bit before the question mark is undefined.

This behaviour might be fine, just wanted to ask.